### DISCUSSION TITLE
Can enableDynamicSizing be toggled frequently while a BottomSheet is open?

### DISCUSSION THREAD

**Discussion by lucasFigueira1** [gorhom/react-native-bottom-sheet#1949] [Q&A]
In my app, I have a map with a BottomSheet that is always open. By default, I want enableDynamicSizing to be off. The BottomSheet has two main states:

A list of nearby contacts (the default content shown when no contact is selected).
A detailed view of a specific contact when a contact (map marker) is pressed.
If I have enableDynamicSizing on by default, and the contact list contains many items, the BottomSheet will snap to the top of the screen when entering the map, which is not desired. However, when a contact is pressed, enableDynamicSizing works perfectly to adjust the sheet size dynamically.

The issue I'm encountering is that when toggling enableDynamicSizing from off to on while the BottomSheet is open, the content changes, but the height of the BottomSheet doesn't adjust dynamically. I suspect this is because a re-render is required for the BottomSheet to update its behavior, but I'm not sure how to implement this properly.

Any kind of help would be much appreciated! 

Regards,
Lucas.

**Comment by pavel-krasnov** [comment]
Try setting a `key` prop to the `BottomSheet` element and reset it every time you want the height to change. This will remount the component.
